lib/util/batch.js

```diff
@@ -129,16 +147,28 @@ function joinQueries(queries, timestamp, batchType, consistencyHierarchy, cqlVer
     , consistencyMap = null
     , hasTimestamp = false
     , hasQuery = false
-    , supportsNestedTimestamp = supportsCql31(cqlVersion)
+    , supportsNestedTimestamp = typeof cqlVersion === 'boolean' ? cqlVersion : supportsCql31(cqlVersion)
```

Comment:
Recursive call, short circuits the version check logic from happening multiple times on nested batches.